### PR TITLE
Adding cosign to build-tools

### PR DESCRIPTION
**Pull request by stewartbutler**
Adding Cosign to the build-tools image to enable image signing.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: docker/build-tools/Dockerfile [hunk 1]
```diff
@@ -271,6 +272,16 @@ RUN set -eux; \
     ${OUTDIR}/usr/local/google-cloud-sdk/bin/gcloud components install alpha --quiet; \
     rm -rf /usr/local/google-cloud-sdk/.install/.backup
 
+# Install cosign (for signing build artifacts) and verify signature
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN set -eux; \
+    ${OUTDIR}/usr/local/google-cloud-sdk/bin/gsutil -q cp gs://cosign-releases/${COSIGN_VERSION}/cosign-linux-amd64 /tmp/cosign \
```
Comment (ericvn) [inline docker/build-tools/Dockerfile:278]: I know we have `TARGETARCH`  as well as checking for x86_64 and aarch64 in various places. This looks to limit us to and/x86_64. Is there an arm version of the code available?

Comment (stewartbutler) [reply docker/build-tools/Dockerfile:278]: There is. I'm adding it.

Comment (stewartbutler) [reply docker/build-tools/Dockerfile:278]: Actually, I take that back -- the only linux variant available is amd64 (see https://github.com/sigstore/cosign/releases/tag/v1.2.1). There's an amd64 and arm variant for mac, but that doesn't matter for this image afaik.

File: docker/build-tools/Dockerfile [hunk 2]
```diff
@@ -71,6 +71,7 @@ ENV BUF_VERSION=v0.43.2
 ENV GCLOUD_VERSION=355.0.0
 ENV KUBETEST2_VERSION=ea531e01c2406c5064aea55fa76a62dbae6da6a1
 ENV BOSKOSCTL_VERSION=c6d730e323f06da1b56dfa14bdab6d7d5dc22e2a
+ENV COSIGN_VERSION=v1.1.0
```
Comment (ericvn) [inline docker/build-tools/Dockerfile:74]: Update to version 1.2.1?

File: docker/build-tools/Dockerfile (hunk 1 above)
Comment (brian-avery) [reply docker/build-tools/Dockerfile:278]: Advantage to this is we can verify the binary before it's installed.

**Review by brian-avery — APPROVED** [review]

**Comment by stewartbutler** [comment]
I'm not sure why this is still failing. It looks like a flake but it's been pretty consistent. Investigating.

File: Makefile.core.mk [hunk 3]
```diff
@@ -38,6 +38,8 @@ containers:
 	@cd docker/build-tools && ./build-and-push.sh
 
 containers-test:
+	# Workaround https://github.com/kubernetes/test-infra/issues/23741. Should be removed once its fixed upstream
+	@iptables -t mangle -A POSTROUTING -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --clamp-mss-to-pmtu
```
Comment (howardjohn) [inline Makefile.core.mk:42]: I don't think its sufficient to do it in the test, the postsubmit job will fail. I would do it in build-and-push

**Review by howardjohn — APPROVED** [review]

**Review by ericvn — APPROVED** [review]
I would make the update as John suggested so the automated build happens (hopefully without incident) and then it looks good.

**Pull request merged**